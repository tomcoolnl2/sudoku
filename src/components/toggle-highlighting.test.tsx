
import { render, screen } from '@testing-library/react'
import { withTheme, withRedux } from '../test'
import { ToggleHighlighting } from '.'


describe('ToggleHighlighting', () => {
	
	it('renders with light mode as default', () => {
		const { container } = render(withRedux(withTheme(<ToggleHighlighting />)))
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should be present within the DOM', () => {
		render(withRedux(withTheme(<ToggleHighlighting />)))
		const buttonElement = screen.getByText(/Enable Highlighting/i)
		expect(buttonElement).toBeInTheDocument()
	})
})