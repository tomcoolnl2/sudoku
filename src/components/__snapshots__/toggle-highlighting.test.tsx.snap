// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleHighlighting should match it's snapshot 1`] = `
<label
  for="highlight-toggle"
>
  Enable Highlighting:
  <input
    checked=""
    name="highlight-toggle"
    type="checkbox"
  />
</label>
`;
